openlibrary/components/MergeUI/MergeTable.vue

```diff
@@ -100,37 +60,40 @@ function fetchRecord(olid) {
         M: 'books',
         A: 'authors'
     }[olid[olid.length - 1]];
-    const endpoint = `/${type}/${olid}.json`;
+    const record_key = `/${type}/${olid}`;
     // FIXME Fetch from prod openlibrary.org, otherwise it's outdated
-    const url = location.host.endsWith('.openlibrary.org') ? `https://openlibrary.org${endpoint}` : endpoint;
-    return fetch(url).then(r => r.json());
+    const url = location.host.endsWith('.openlibrary.org') ? `https://openlibrary.org${record_key}.json` : `${record_key}.json`;
+    return fetch(url).then(r => {
+        return (r.ok) ? r.json() : {key: record_key, type: {key: '/type/none'}, error: r.statusText};
+    });
 }
 
 export default {
     name: 'MergeTable',
     components: {
-        EditionSnippet,
         MergeRow
     },
     data() {
-        const sorted = _.sortBy(this.olids, olid =>
-            parseFloat(olid.match(/\d+/)[0])
-        );
         return {
-            master_key: `/works/${sorted[0]}`,
+            master_key: null,
             /** @type {{[key: string]: Boolean}} */
-            selected: _.fromPairs(this.olids.map(olid => [`/works/${olid}`, true]))
+            selected: []
         };
     },
     props: {
-        olids: Array
+        olids: Array,
+        show_diffs: Boolean
     },
     asyncComputed: {
         async records() {
-            const sorted = _.sortBy(this.olids, olid =>
+            const olids_sorted = _.sortBy(this.olids, olid =>
                 parseFloat(olid.match(/\d+/)[0])
             );
-            return await Promise.all(sorted.map(fetchRecord));
+            const records_unsorted = await Promise.all(olids_sorted.map(fetchRecord));
+            const records = _.orderBy(records_unsorted, record => record.type.key, 'desc');
+            this.master_key = records[0].key
```

Comment:
If so this is a little clearer; the use of sorted things but then not sorted but then sorted again is a little confusing.
```suggestion
            const records = await Promise.all(olids_sorted.map(fetchRecord));
            // Since we can also have orphaned editions; ensure a work is the primary
            this.master_key = records.filter(r => r.type.key = '/type/work')[0].key || records[0].key;
```

(Note: `.find` would be better JS, but would need a `?.key`, and I can't recall if we can use `?.` in the Vue code)